### PR TITLE
React native upgrade 0.62.1

### LINKED ISSUE #1201: Migrate to React Native v0.61.+
Sooner or later we will get pay the price of being behind and the migration itself it's a considerably big project.

Things to take into account:

- Need to create a new project and start moving things in.
- Make sure packages support pods instead of react-native link
- Try hermes and compare with current performance of v8

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

This PR addresses the upgrade from React Native 0.59.10 to 0.62.1. Although there were lots of changes necessary, an effort was made in order to keep as much as possible the version of the libraries that were already being used (**no unnecessary library upgrades were made**), and also to keep as much of the code as intact as possible (**no unnecessary code changes were made**). This makes it easier to track any possible problems to the React Native upgrade itself. The upgrading of libraries to a newer current version is a step 2 outside of this PR context.

**Note**: To run this PR, don't forget to check the changes on the README. More specifically installing the pods:
`brew install cocoapods`
And on the root of the project
`cd ios && pod install && cd ..`

To open the project on Xcode, the file **MetaMask.xcworkspace** should be used instead of MetaMask.xcodeproj.

`yarn clean` also installs the pods now.

## General file changes
The first part of the upgrade was making all the file changes needed in order to integrate the new React Native version. Those changes can be followed on this [upgrade-helper](https://react-native-community.github.io/upgrade-helper/?from=0.59.10&to=0.62.1).

## Library changes
One of the biggest changes in React 0.60+ is the integration by default of the [pods system](https://reactnative.dev/blog/2019/07/03/version-60#cocoapods-by-default). This means every library needed to be unlinked. This is because with the new system, every library is [automatically linked](https://reactnative.dev/blog/2019/07/03/version-60#native-modules-are-now-autolinked). More specifically, the **libraries that were unlinked** were:
- react-native-sensors
- react-native-reanimated
- react-native-webview
- @react-native-community/netinfo
- react-native-view-shot
- lottie-react-native
- @react-native-community/async-storage
- react-native-push-notification
- react-native-background-timer
- react-native-device-info
- react-native-svg
- react-native-gesture-handler
- react-native-screens
- react-native-branch
- react-native-camera
- react-native-fs
- react-native-i18n
- react-native-keychain
- react-native-share
- react-native-randombytes
- react-native-vector-icons

### Libraries that did not support auto-linking
Some libraries couldn't be unlinked because they did not support auto-linking. Furthermore, in order for some of them to work, changes needed to be made on each library.

More specifically on iOS:
- react-native-aes-crypto. [See changes here.](https://github.com/MetaMask/react-native-aes/commit/b73625d7e1ce711ed3c2fbd795eea2bda5a5e9ad)
- react-native-os. [See  changes here.](https://github.com/MetaMask/react-native-os/commit/67cbc5787295f36575c64442e7f74b803b3de8dd)
- react-native-search-api. [See changes here.](https://github.com/MetaMask/react-native-search-api/commit/270512b6641bfaba155ca8b31aa2e1d8b5cf0ec6)
- react-native-tcp [See changes here.](https://github.com/MetaMask/react-native-tcp/commit/961328b678b01427b0d2cd7ddb506f620a5418e1)
- react-native-scrollable-tab-view. The fix for this library was actually **bumping the version**, since the new version supported auto-linking and it seemed to already integrate our custom changes. Version bumped from [our custom version](https://github.com/estebanmino/react-native-scrollable-tab-view/commit/f954274c6260a07c956e2185c611aed24b2f1526) to [original v1.0.0](https://github.com/ptomasroos/react-native-scrollable-tab-view/tree/v1.0.0)

and on Android:
- lottie-react-native
- react-native-gesture-handler

### Patches removed
- `react-native+0.59.10.patch`. Since a new version of React Native was installed, this patch did not make sense anymore. No patch for the new version was necessary, at least for successfully compiling and running the app.

### Removed assets
- Removed fonts: Because of the auto-linking features, all the fonts that were manually added to Xcode from `react-native-vector-icons` were removed. The auto-linking should add those fonts automatically now.

## Errors fixed
After the app compiled successfully, there were some errors encountered when running the app. More specifically:
- App crashed on Android on any screen with our `WebView`. After some research, [the fix was modifying our custom WebView.](https://github.com/MetaMask/react-native-webview/commit/408df176e6fe7c00aa05c2fddb4054968330e8a1)
- App crashed when closing the side nav drawer on iOS release. The fix was adding `import 'react-native-gesture-handler';` to the` index.js file`. See [here](https://github.com/MetaMask/metamask-mobile/compare/react-native-upgrade-0.62.1-wip#diff-168726dbe96b3ce427e7fedce31bb0bcR3).
- Metro bundler did not work with the new React Native version. The metro version needed to be upgraded from 0.55.0 to 0.59.0.

## Warnings fixed

- `Animated: 'useNativeDriver' was not specified. This is a required 'option and must be explicitly set to true or false`. This warning was fixed by **upgrading** `react-native-modal` from 9.0.0 to 11.5.6.
- `Animated.event now requires a second argument for options`. This warning was fixed by **upgrading** `react-native-modal` from 9.0.0 to 11.5.6.
- `Calling getNode() on the ref of an Animated component is no longer necessary`. Since a lot of libraries were causing this warning like `react-native-scrollable-tab-view`, `react-native-keyboard-aware-scroll-view` and `react-native-tab-view`, **this warning was added to the ignore list**. We should address this by upgrading the libraries mentioned.
- `componentWillUpdate deprecated`. Since this appears on a lot of our components, **this warning was added to the ignore list**. Check [this](https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate) for possible fix.
- `componentWillReceiveProps deprecated`. Since this appears on a lot of our components, **this warning was added to the ignore list**. Check [this](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) for possible fix.

## Lint

The lint plugin we used called `eslint-config-react-native` no longer worked correctly with the new React Native version. The reccommended plugin now is `@react-native-community/eslint-config`. An effort was made so that our current rules remained the same. Because of this, there were lots of rules added so a new file called `.eslintrc.js` was created to accommodate all the rules. It's possible that this new config is more strict, so we may still change the config file to our liking.

**Issue**

Resolves #1201

### CODE REVIEW

File: android/.settings/org.eclipse.buildship.core.prefs [hunk 1]
```diff
@@ -1,2 +1,13 @@
+arguments=
+auto.sync=false
+build.scans.enabled=false
+connection.gradle.distribution=GRADLE_DISTRIBUTION(WRAPPER)
 connection.project.dir=
 eclipse.preferences.version=1
+gradle.user.home=
+java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_171.jdk/Contents/Home
```
Comment (rickycodes) [inline android/.settings/org.eclipse.buildship.core.prefs:8]: Do we want to add this? My thinking here is that this is system dependant and will be different for everyone. What if a user has Java installed in a different directory? I'm on linux and so, I don't even have a `/Library` folder on my system (as an example). Of course, i also don't use Eclipse either so :shrug:

File: package.json [hunk 2]
```diff
@@ -1,334 +1,302 @@
...
-    "net": "react-native-tcp",
-    "fs": "react-native-level-fs"
-  },
-  "rnpm": {
-    "assets": [
-      "./app/fonts"
-    ]
-  }
-}
+	"name": "metamask",
```
Comment (rickycodes) [inline package.json:2]: I'm curious why we're re tabbing this? It's difficult to get a sense of what actually changed here.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
+module.exports = {
+	root: true,
```
Comment (rickycodes) [inline .eslintrc.js:2]: maybe pulling the `eslintConfig` into a `.eslintrc.js` file (a change I'm in favour of) could be done in a separate PR to keep changes minimal in this one?

File: android/.settings/org.eclipse.buildship.core.prefs (hunk 1 above)
Comment (andrepimenta) [reply android/.settings/org.eclipse.buildship.core.prefs:8]: Yes I think you are right, I am not sure why this file was created. Thanks for pointing it out!

File: package.json (hunk 2 above)
Comment (andrepimenta) [reply package.json:2]: Yes this is strange, will fix it!

File: .eslintrc.js (hunk 3 above)
Comment (andrepimenta) [reply .eslintrc.js:2]: The problem is that there is a huge amount of rules now, package.json would be very confusing. I think we have no choice than to do this either way.

**Pull request closed without merging**